config/karma.config.js

```diff
@@ -3,11 +3,12 @@
 
 import webpackConfig from './webpack.config';
 
-var coverageReporters = [{
+const coverageReporters = [{
   type: 'text-summary',
 }];
 
-var newWebpackConfig = Object.assign({}, webpackConfig, {
+// eslint-disable-next-line prefer-const
+let newWebpackConfig = Object.assign({}, webpackConfig, {
```

Comment:
Coming from the Ruby world, I always feel like defining something as a constant is more about signalling intent than enforcing further assignment rules.

I like the idea of saying "we're defining X as a constant because we make no changes to it". If we're going to change the contents of a variable after we define it, we should signal that with var/let.

Being able to mess with constants isn't obvious (as evidenced by this discussion), even if it's technically correct (the best kind of correct 😉)

- tofumatt

> On 5 Mar 2016, at 00:24, Mark Striemer <notifications@github.com> wrote:
> 
> In config/karma.config.js:
> 
> >    type: 'text-summary',
> >  }];
> >  
> > -var newWebpackConfig = Object.assign({}, webpackConfig, {
> > +// eslint-disable-next-line prefer-const
> > +let newWebpackConfig = Object.assign({}, webpackConfig, {
> I'm not sure eslint is getting it "wrong" here. const just means the value can't be reassigned, not that it can't be mutated. I think const would be needed for immutability but it doesn't enforce immutability (the type also needs to be immutable).
> 
> I think this should still be const even though it gets mutated. I'm fine with it as is though, it can get changed if someone feels like changing it.
> 
> These articles have some mildly interesting points: I did a search and found an article called ES6 const is not about immutability which goes into that and someone posted another article in the comments there about why you shouldn't use const by default, which is interesting but I'm not entirely convinced.
> 
> —
> Reply to this email directly or view it on GitHub.
>